frontend/test/src/LiveMessageManager.spec.js

```diff
@@ -0,0 +1,73 @@
+import { expect } from 'chai'
+import {
+  LiveMessageManager,
+  LIVE_MESSAGE_STATUS
+} from '../../src/util/LiveMessageManager.js'
+
+const apiUrl = 'http://localhost/api'
+const userId = 1
+
+const openedManager = (heartbeatTimeOut) => {
+  const manager = new LiveMessageManager(heartbeatTimeOut)
+  manager.openLiveMessageConnection(userId, apiUrl)
+  const mockEventSource = manager.eventSource
+  mockEventSource.emitOpen()
+  return { manager, mockEventSource }
+}
+
+describe('LiveMessageManager class', () => {
+  const managers = []
+
+  describe('the openLiveMessageConnection() method', () => {
+    const manager = new LiveMessageManager()
+    managers.push(manager)
+    manager.openLiveMessageConnection(userId, apiUrl)
+    const mockEventSource = manager.eventSource
+    it('should set the create an EventSource and set manager status to PENDING when called', () => {
+      expect(mockEventSource).to.not.be.equal(null)
+      expect(mockEventSource.readyState).to.be.equal(0)
+      expect(manager.status).to.be.equal(LIVE_MESSAGE_STATUS.PENDING)
+    })
+    it('should set manager status to OPENED when initial event is received', () => {
+      mockEventSource.emitOpen()
+      mockEventSource.emit('stream-open')
+      expect(mockEventSource.readyState).to.be.equal(1)
+      expect(manager.status).to.be.equal(LIVE_MESSAGE_STATUS.OPENED)
+    })
+    it('should restart connection when an error is raised by EventSource', () => {
+      mockEventSource.emitError()
+      expect(manager.status).to.be.equal(LIVE_MESSAGE_STATUS.PENDING)
+    })
+  })
+
+  describe('the closeLiveMessageConnection() method', () => {
+    it('should close the EventSource', () => {
+      const { manager, mockEventSource } = openedManager()
+      managers.push(manager)
+      manager.closeLiveMessageConnection()
+      expect(mockEventSource.readyState).to.be.equal(2)
+    })
+    it('should set manager status to CLOSED', () => {
+      const { manager } = openedManager()
+      managers.push(manager)
+      manager.closeLiveMessageConnection()
+      expect(manager.status).to.be.equal(LIVE_MESSAGE_STATUS.CLOSED)
+    })
+  })
+
+  describe('the heartbeat timer', () => {
+    it('should restart connection when the heartbeat event is not received in time', async () => {
```

Comment:
Just a question, Do you expect heartbeat event at a regular time ?
If so, you should probably modify a bit https://github.com/tracim/tracim/blob/1e9d9c30fd4e7e2de62888b0b81d83de358cc005/backend/tracim_backend/views/core_api/user_controller.py#L626
with "interval"
see https://pushpin.org/docs/advanced/#keep-alives

Reply:
Ah, yes you're right I saw this… I can also reset the frontend timer when a tlm is received :thinking: